test/tandem/testTandemSimulator.js

```diff
@@ -0,0 +1,608 @@
+/*
+ * == BSD2 LICENSE ==
+ * Copyright (c) 2015, Tidepool Project
+ *
+ * This program is free software; you can redistribute it and/or modify it under
+ * the terms of the associated License, which is identical to the BSD 2-Clause
+ * License as published by the Open Source Initiative at opensource.org.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the License for more details.
+ *
+ * You should have received a copy of the License along with this program; if
+ * not, you can obtain one from Tidepool Project at tidepool.org.
+ * == BSD2 LICENSE ==
+ */
+
+/* global beforeEach, describe, it */
+
+var _ = require('lodash');
+var expect = require('salinity').expect;
+
+var pwdSimulator = require('../../lib/tandem/tandemSimulator.js');
+var builder = require('../../lib/objectBuilder')();
+
+describe('tandemSimulator.js', function() {
+  var simulator = null;
+
+  beforeEach(function(){
+    simulator = pwdSimulator.make();
+  });
+
+  describe('smbg', function(){
+    it('passes through', function(){
+      var val = {
+        time: '2014-09-25T01:00:00.000Z',
+        deviceTime: '2014-09-25T01:00:00',
+        timezoneOffset: 0,
+        conversionOffset: 0,
+        deviceId: 'InsOmn1234',
+        units: 'mg/dL',
+        type: 'smbg',
+        value: 1.3
+      };
+
+      simulator.smbg(val);
+      expect(simulator.getEvents()).deep.equals([val]);
+    });
+  });
+
+  describe('bolus', function(){
+    describe('normal', function() {
+      var val = {
+        time: '2014-09-25T01:00:00.000Z',
+        deviceTime: '2014-09-25T01:00:00',
+        timezoneOffset: 0,
+        conversionOffset: 0,
+        deviceId: 'InsOmn1234',
+        normal: 1.3,
+        type: 'bolus',
+        subType: 'normal'
+      };
+
+      it('passes through', function(){
+        simulator.bolus(val);
+        expect(simulator.getEvents()).deep.equals([val]);
+      });
+
+      it('does not pass through a zero-volume bolus that does not have an expectedNormal', function() {
```

Comment:
I'm not sure it's possible to have an extended bolus where `bolex_insulin_requested` is not defined, as the pump requires you to put in at least 0.4u?